pkg/registry/populator.go

```diff
@@ -23,18 +23,22 @@ type Dependencies struct {
 
 // DirectoryPopulator loads an unpacked operator bundle from a directory into the database.
 type DirectoryPopulator struct {
-	loader      Load
-	graphLoader GraphLoader
-	querier     Query
-	imageDirMap map[image.Reference]string
+	loader          Load
+	graphLoader     GraphLoader
+	querier         Query
+	imageDirMap     map[image.Reference]string
+	overwriteDirMap map[string]map[image.Reference]string
+	overwrite       bool
 }
 
-func NewDirectoryPopulator(loader Load, graphLoader GraphLoader, querier Query, imageDirMap map[image.Reference]string) *DirectoryPopulator {
+func NewDirectoryPopulator(loader Load, graphLoader GraphLoader, querier Query, imageDirMap map[image.Reference]string, overwriteDirMap map[string]map[image.Reference]string, overwrite bool) *DirectoryPopulator {
 	return &DirectoryPopulator{
-		loader:      loader,
-		graphLoader: graphLoader,
-		querier:     querier,
-		imageDirMap: imageDirMap,
+		loader:          loader,
+		graphLoader:     graphLoader,
+		querier:         querier,
+		imageDirMap:     imageDirMap,
+		overwriteDirMap: overwriteDirMap,
```

Comment:
do we need to distinguish which ones are "overwrites" up front?

Reply:
yeah I think it makes sense to do it upfront so we only pull the images when we need to. I think more of the registry package needs to be made public in order to make that possible but it shouldn't be too bad